packages/enzyme-adapter-react-16/src/detectFiberTags.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+function getFiber(element) {
+  const container = global.document.createElement('div');
+  let inst = null;
+  class Tester extends React.Component {
+    render() {
+      inst = this;
+      return element;
+    }
+  }
+  ReactDOM.render(React.createElement(Tester), container);
+  return inst._reactInternalFiber.child;
+}
+
+function detectFiberTags() {
+  const supportsMode = typeof React.StrictMode !== 'undefined';
+  const supportsContext = typeof React.createContext !== 'undefined';
+  const supportsForwardRef = typeof React.forwardRef !== 'undefined';
```

Comment:
These assume you're not going to use a newer `react` minor with an older `react-dom` minor. If that's not an assumption we're comfortable making, I can add some try-catches and fall back.

Reply:
I think that it's fine if things break with a mismatched react/react-dom - those have always had to match minor versions to work properly.